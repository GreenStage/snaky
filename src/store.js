import {createStore} from "redux";

const store = createStore((st = {}, a) => {
    switch (a.type) {
        case 'UPDATE_SCORE':
            return {...st, score: a.score};
        default: return st;
    }
});

export default store;