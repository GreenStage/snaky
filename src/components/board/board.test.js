import React from 'react'
import {createStore} from 'redux'
import {render, fireEvent, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Board from "./index";
import {Provider} from "react-redux";
import store from "../../store";

describe('board tests', () => {
    test('renders correctly', () => {
        const {container} = render(
            <Provider store={store}>
                <Board/>
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });

    test('updates counter on score update', () => {
        const {container} = render(
            <Provider store={store}>
                <Board/>
            </Provider>
        );

        const scoreEl = container.querySelector('[id="score"]');
        expect(scoreEl).toBeTruthy();
        store.dispatch({
            type: 'UPDATE_SCORE',
            score: '22',
        });
        expect(scoreEl).toHaveTextContent(22);
    });
})