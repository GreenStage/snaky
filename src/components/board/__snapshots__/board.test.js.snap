// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`board tests renders correctly 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="makeStyles-header-1"
    >
      <img
        alt="snaky"
        class="makeStyles-snake-2"
        src="snake.png"
      />
      <h1>
        Snaky
      </h1>
      v0.1
    </div>
    <p
      class="makeStyles-score-3"
      id="score"
    />
  </div>
</div>
`;
